### PR TITLE
[CFGPrinter] Allow CFG dumps with a given filename

### PR DESCRIPTION
**Pull request by LewisCrawford**
Add functions to print the CFG via the debugger using a specified filename.

This is useful when comparing CFGs for the same function before vs after a change, or when handling functions with names that are too long to be file names.

### CODE REVIEW

**Comment by LewisCrawford** [comment]
Requesting review from @WenleiHe and @david-xl , as they reviewed a similar CFGPrinter debugger utils patch a few years ago in https://reviews.llvm.org/D77978

File: llvm/include/llvm/IR/Function.h [hunk 1]
```diff
@@ -942,9 +942,14 @@ class LLVM_ABI Function : public GlobalObject, public ilist_node<Function> {
...
 
+  /// viewCFG - This function is meant for use from the debugger. It works just
+  /// like viewCFG(), but generates the dot file with the given filename.
+  void viewCFG(const char *Name) const;
+
   /// Extended form to print edge weights.
   void viewCFG(bool ViewCFGOnly, const BlockFrequencyInfo *BFI,
-               const BranchProbabilityInfo *BPI) const;
+               const BranchProbabilityInfo *BPI,
+               const char *Name = nullptr) const;
```
Comment (WenleiHe) [inline llvm/include/llvm/IR/Function.h:952]: nit: `Name` -> `outputFileName` to be explicit.

**Review by WenleiHe — APPROVED** [review]
lgtm with a nit. thanks.

File: llvm/include/llvm/IR/Function.h (hunk 1 above)
Comment (LewisCrawford) [reply llvm/include/llvm/IR/Function.h]: Thanks for the review!
I've changed it to OutputFileName now.

**Pull request merged**